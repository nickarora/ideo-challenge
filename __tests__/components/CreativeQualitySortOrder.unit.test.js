const sortByName = jest.fn()
const sortByScore = jest.fn()

const mockSortConfig = sortConfig => {
  jest.doMock('react-redux', () => {
    return {
      connect: () => Component => () => (
        <Component
          sortConfig={sortConfig}
          sortByName={sortByName}
          sortByScore={sortByScore}
        />
      ),
    }
  })

  return require('~/components/CreativeQualitySortOrder').default // eslint-disable-line
}

beforeEach(() => {
  jest.resetModules()
})

describe('CreativeQualitySortOrder', () => {
  describe('when SortOrder is by score', () => {
    it('renders as ascending', () => {
      const CreativeQualitySortOrder = mockSortConfig({
        sortOrder: 'SCORE',
        scoreAscending: true,
        nameAscending: false,
      })

      const wrapper = shallow(<CreativeQualitySortOrder />).dive()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders as descending', () => {
      const CreativeQualitySortOrder = mockSortConfig({
        sortOrder: 'SCORE',
        scoreAscending: false,
        nameAscending: false,
      })

      const wrapper = shallow(<CreativeQualitySortOrder />).dive()
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when SortOrder is by name', () => {
    it('renders as ascending', () => {
      const CreativeQualitySortOrder = mockSortConfig({
        sortOrder: 'NAME',
        scoreAscending: false,
        nameAscending: true,
      })

      const wrapper = shallow(<CreativeQualitySortOrder />).dive()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders as descending', () => {
      const CreativeQualitySortOrder = mockSortConfig({
        sortOrder: 'NAME',
        scoreAscending: false,
        nameAscending: false,
      })

      const wrapper = shallow(<CreativeQualitySortOrder />).dive()
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('selects score on tap', () => {
    const CreativeQualitySortOrder = mockSortConfig({
      sortOrder: 'NAME',
      scoreAscending: true,
      nameAscending: false,
    })

    const wrapper = shallow(<CreativeQualitySortOrder />).dive()
    wrapper.find('#sort-by-score').simulate('click')
    expect(sortByScore).toHaveBeenCalled()
  })

  it('renders as descending', () => {
    const CreativeQualitySortOrder = mockSortConfig({
      sortOrder: 'NAME',
      scoreAscending: false,
      nameAscending: false,
    })

    const wrapper = shallow(<CreativeQualitySortOrder />).dive()
    wrapper.find('#sort-by-name').simulate('click')
    expect(sortByName).toHaveBeenCalled()
  })
})
