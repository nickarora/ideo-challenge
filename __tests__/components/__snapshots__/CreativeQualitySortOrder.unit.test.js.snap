// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreativeQualitySortOrder when SortOrder is by name renders as ascending 1`] = `
<styled.section>
  <styled.button
    className="btn btn-primary false"
    id="sort-by-score"
    onClick={[MockFunction]}
  >
    Sort by score 
    ↓
  </styled.button>
  <styled.button
    className="btn btn-primary btn-info"
    id="sort-by-name"
    onClick={[MockFunction]}
  >
    Sort by name 
    ↑
  </styled.button>
</styled.section>
`;

exports[`CreativeQualitySortOrder when SortOrder is by name renders as descending 1`] = `
<styled.section>
  <styled.button
    className="btn btn-primary false"
    id="sort-by-score"
    onClick={[MockFunction]}
  >
    Sort by score 
    ↓
  </styled.button>
  <styled.button
    className="btn btn-primary btn-info"
    id="sort-by-name"
    onClick={[MockFunction]}
  >
    Sort by name 
    ↓
  </styled.button>
</styled.section>
`;

exports[`CreativeQualitySortOrder when SortOrder is by score renders as ascending 1`] = `
<styled.section>
  <styled.button
    className="btn btn-primary btn-info"
    id="sort-by-score"
    onClick={[MockFunction]}
  >
    Sort by score 
    ↑
  </styled.button>
  <styled.button
    className="btn btn-primary false"
    id="sort-by-name"
    onClick={[MockFunction]}
  >
    Sort by name 
    ↓
  </styled.button>
</styled.section>
`;

exports[`CreativeQualitySortOrder when SortOrder is by score renders as descending 1`] = `
<styled.section>
  <styled.button
    className="btn btn-primary btn-info"
    id="sort-by-score"
    onClick={[MockFunction]}
  >
    Sort by score 
    ↓
  </styled.button>
  <styled.button
    className="btn btn-primary false"
    id="sort-by-name"
    onClick={[MockFunction]}
  >
    Sort by name 
    ↓
  </styled.button>
</styled.section>
`;
